io/zenoh-links/zenoh-link-quic/src/utils.rs

```diff
@@ -160,40 +158,40 @@ impl TlsServerConfig {
         let tls_server_private_key = TlsServerConfig::load_tls_private_key(config).await?;
         let tls_server_certificate = TlsServerConfig::load_tls_certificate(config).await?;
 
-        let certs: Vec<Certificate> =
+        let certs: Vec<CertificateDer> =
             rustls_pemfile::certs(&mut Cursor::new(&tls_server_certificate))
-                .map_err(|err| zerror!("Error processing server certificate: {err}."))?
-                .into_iter()
-                .map(Certificate)
-                .collect();
+                .collect::<Result<_, _>>()
+                .map_err(|err| zerror!("Error processing server certificate: {err}."))?;
 
-        let mut keys: Vec<PrivateKey> =
+        let mut keys: Vec<PrivateKeyDer> =
             rustls_pemfile::rsa_private_keys(&mut Cursor::new(&tls_server_private_key))
-                .map_err(|err| zerror!("Error processing server key: {err}."))?
-                .into_iter()
-                .map(PrivateKey)
-                .collect();
+                .map(|x| x.map(PrivateKeyDer::from))
+                .collect::<Result<_, _>>()
+                .map_err(|err| zerror!("Error processing server key: {err}."))?;
 
         if keys.is_empty() {
             keys = rustls_pemfile::pkcs8_private_keys(&mut Cursor::new(&tls_server_private_key))
-                .map_err(|err| zerror!("Error processing server key: {err}."))?
-                .into_iter()
-                .map(PrivateKey)
-                .collect();
+                .map(|x| x.map(PrivateKeyDer::from))
+                .collect::<Result<_, _>>()
+                .map_err(|err| zerror!("Error processing server key: {err}."))?;
         }
 
         if keys.is_empty() {
             keys = rustls_pemfile::ec_private_keys(&mut Cursor::new(&tls_server_private_key))
-                .map_err(|err| zerror!("Error processing server key: {err}."))?
-                .into_iter()
-                .map(PrivateKey)
-                .collect();
+                .map(|x| x.map(PrivateKeyDer::from))
+                .collect::<Result<_, _>>()
+                .map_err(|err| zerror!("Error processing server key: {err}."))?;
         }
 
         if keys.is_empty() {
             bail!("No private key found for TLS server.");
         }
 
+        // Install rustls provider
+        rustls::crypto::ring::default_provider()
+            .install_default()
+            .ok();
```

Comment:
I think it's better to return an error if this operation fails.
```suggestion
            .map_err(|err| zerror!("Error installing TLS crypto provider: {err}."))?;
```